### PR TITLE
vtk: update to 9.2.2

### PR DESCRIPTION
**Pull request by ar-jan**
#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

**Comment by kruceter** [comment]
Failing build for x86_64-musl (possibly?) can be fixed with this patch:

```diff
diff --color -u -r VTK-9.2.2-src/ThirdParty/ioss/vtkioss/Ioss_FileInfo.C VTK-9.2.2/ThirdParty/ioss/vtkioss/Ioss_FileInfo.C
--- VTK-9.2.2-src/ThirdParty/ioss/vtkioss/Ioss_FileInfo.C	2022-11-10 19:57:10.104509831 +0300
+++ VTK-9.2.2/ThirdParty/ioss/vtkioss/Ioss_FileInfo.C	2022-11-10 21:30:44.327591097 +0300
@@ -26,7 +26,11 @@
 #define S_ISDIR(m) (((m)&_S_IFMT) == _S_IFDIR)
 #endif
 #else
+#ifdef __GLIBC__
 #include <sys/unistd.h>
+#else
+#include <unistd.h>
+#endif
 #endif
 
 #ifdef SEACAS_HAVE_MPI
@@ -35,9 +39,6 @@
 
 #include <cstdio>
 #include <sys/stat.h>
-#ifndef _MSC_VER
-#include <unistd.h>
-#endif
 
 namespace {
   bool internal_access(const std::string &name, int mode);
```

I see that the patch fixing build with GCC 12 has been removed, meaning that vtk has to be tested against GCC 12 once again.

**Comment by ar-jan** [comment]
Thanks for the x86_64-musl patch! The gcc12 patch is [no longer needed](https://gitlab.kitware.com/vtk/vtk/-/commit/e066c3f).

**Comment by paper42** [comment]
`/usr/include/sys/unistd.h` just contains `#include <unistd.h>` on glibc, so I think there is no need to put that behind an ifdef, we can just include it everywhere. @kruceter could you send that patch upstream if they don't have it yet?

**Comment by ar-jan** [comment]
Tested a few [VTK Python examples](https://kitware.github.io/vtk-examples/site/Python/) and confirmed that freecad works as well. I don't otherwise use VTK but all seems OK. Anything more needed here?

**Comment by paper42** [comment]
> `/usr/include/sys/unistd.h` just contains `#include <unistd.h>` on glibc, so I think there is no need to put that behind an ifdef, we can just include it everywhere. @kruceter could you send that patch upstream if they don't have it yet?

Could you use <https://github.com/sandialabs/seacas/pull/348.patch>?

**Comment by paper42** [comment]
> > `/usr/include/sys/unistd.h` just contains `#include <unistd.h>` on glibc, so I think there is no need to put that behind an ifdef, we can just include it everywhere. @kruceter could you send that patch upstream if they don't have it yet?
> 
> Could you use [sandialabs/seacas@`348`.patch](https://github.com/sandialabs/seacas/pull/348.patch)?

I meant including the header, download the patch from that link and put it to the right directory.

File: srcpkgs/vtk/template [hunk 1]
```diff
@@ -22,7 +26,7 @@ maintainer="Piraty <piraty1@inbox.ru>"
 license="BSD-3-Clause"
 homepage="https://www.vtk.org"
 distfiles="https://www.vtk.org/files/release/${version:0:3}/VTK-${version}.tar.gz"
-checksum=1b39a5e191c282861e7af4101eaa8585969a2de05f5646c9199a161213a622c7
+checksum=1c5b0a2be71fac96ff4831af69e350f7a0ea3168981f790c000709dcf9121075
 nocross="hdf5 is nocross"
```
Comment (paper42) [inline srcpkgs/vtk/template:30]: I think that if a dependency is nocross, the package should just not build if xbps-src can not build that dependency, so this should not be necessary.

**Comment by ar-jan** [comment]
Upstream patch does not apply to the version vendored with vtk-9.2.2.

File: srcpkgs/vtk/template (hunk 1 above)
Comment (ar-jan) [reply srcpkgs/vtk/template:30]: If I try building locally with `./xbps-src pkg -a 86_64-musl vtk` I get `ERROR: XBPS_CROSS_TRIPLET is not defined!`. Does that mean we should leave it as is?

Comment (paper42) [reply srcpkgs/vtk/template:30]: That's a wrong arch, you missed an x.

Comment (ar-jan) [reply srcpkgs/vtk/template:30]: lol oops it's getting late :)

Comment (ar-jan) [reply srcpkgs/vtk/template:30]: Now for realz: `CMake Error at CMake/vtkCrossCompiling.cmake:7 (find_package):   By not providing "FindVTKCompileTools.cmake" in CMAKE_MODULE_PATH this project has asked CMake to find a package configuration file provided by "VTKCompileTools", but CMake did not find one.`

Comment (kruceter) [reply srcpkgs/vtk/template:30]: I have removed `nocross` on my side and it definitely cannot crossbuild due to hdf5's templated containing `nocross` as well.

Just like @paper42 said, if a dependency is nocross, the package should not build at all.

It is safe to remove the line with `nocross` here.

File: common/shlibs [hunk 2]
```diff
@@ -3259,110 +3259,116 @@ libgnuradio-pdu.so.3.10.3 gnuradio-3.10.3.0_1
...
+libvtkRenderingSceneGraph-9.2.so.1 vtk-9.2.2_1
+libvtkRenderingUI-9.2.so.1 vtk-9.2.2_1
+libvtkRenderingVolume-9.2.so.1 vtk-9.2.2_1
+libvtkRenderingVolumeOpenGL2-9.2.so.1 vtk-9.2.2_1
+libvtkRenderingVtkJS-9.2.so.1 vtk-9.2.2_1
+libvtksys-9.2.so.1 vtk-9.2.2_1
+libvtkTestingRendering-9.2.so.1 vtk-9.2.2_1
+libvtkViewsContext2D-9.2.so.1 vtk-9.2.2_1
+libvtkViewsCore-9.2.so.1 vtk-9.2.2_1
+libvtkViewsInfovis-9.2.so.1 vtk-9.2.2_1
```
Comment (kruceter) [inline common/shlibs:3371]:
```suggestion
libvtkViewsInfovis-9.2.so.1 vtk-9.2.2_1
libvtkWrappingTools-9.2.so.1 vtk-9.2.2_1
libvtkloguru-9.2.so.1 vtk-9.2.2_1
libvtkpugixml-9.2.so.1 vtk-9.2.2_1
libvtkfmt-9.2.so.1 vtk-9.2.2_1
libvtkdoubleconversion-9.2.so.1 vtk-9.2.2_1
libvtklz4-9.2.so.1 vtk-9.2.2_1
libvtklzma-9.2.so.1 vtk-9.2.2_1
libvtkzlib-9.2.so.1 vtk-9.2.2_1
libvtkexpat-9.2.so.1 vtk-9.2.2_1
libvtkParallelDIY-9.2.so.1 vtk-9.2.2_1
libvtkfreetype-9.2.so.1 vtk-9.2.2_1
libvtkDICOMParser-9.2.so.1 vtk-9.2.2_1
libvtkjpeg-9.2.so.1 vtk-9.2.2_1
libvtkmetaio-9.2.so.1 vtk-9.2.2_1
libvtkpng-9.2.so.1 vtk-9.2.2_1
libvtktiff-9.2.so.1 vtk-9.2.2_1
libvtkglew-9.2.so.1 vtk-9.2.2_1
libvtkhdf5-9.2.so.1 vtk-9.2.2_1
libvtkhdf5_hl-9.2.so.1 vtk-9.2.2_1
libvtkogg-9.2.so.1 vtk-9.2.2_1
libvtktheora-9.2.so.1 vtk-9.2.2_1
libvtknetcdf-9.2.so.1 vtk-9.2.2_1
libvtksqlite-9.2.so.1 vtk-9.2.2_1
libvtklibproj-9.2.so.1 vtk-9.2.2_1
libvtkjsoncpp-9.2.so.1 vtk-9.2.2_1
libvtklibxml2-9.2.so.1 vtk-9.2.2_1
libvtkcgns-9.2.so.1 vtk-9.2.2_1
libvtkexodusII-9.2.so.1 vtk-9.2.2_1
libvtkioss-9.2.so.1 vtk-9.2.2_1
libvtklibharu-9.2.so.1 vtk-9.2.2_1
libvtkgl2ps-9.2.so.1 vtk-9.2.2_1
libvtkverdict-9.2.so.1 vtk-9.2.2_1
```

**Review by kruceter — CHANGES_REQUESTED** [review]

File: common/shlibs [hunk 3]
```diff
@@ -3259,110 +3259,148 @@ libgnuradio-pdu.so.3.10.3 gnuradio-3.10.3.0_1
...
+libvtkIOXML-9.2.so.1 vtk-9.2.2_1
+libvtkIOXMLParser-9.2.so.1 vtk-9.2.2_1
+libvtkjpeg-9.2.so.1 vtk-9.2.2_1
+libvtkjsoncpp-9.2.so.1 vtk-9.2.2_1
+libvtkkissfft-9.2.so.1 vtk-9.2.2_1
+libvtklibharu-9.2.so.1 vtk-9.2.2_1
+libvtklibproj-9.2.so.1 vtk-9.2.2_1
+libvtklibxml2-9.2.so.1 vtk-9.2.2_1
+libvtkloguru-9.2.so.1 vtk-9.2.2_1
+libvtklz4-9.2.so.1 vtk-9.2.2_1
```
Comment (paper42) [inline common/shlibs:3367]: Is this required by some package? If not, we should not include it in shlibs and keep this as small as possible.

Comment (ar-jan) [reply common/shlibs:3367]: No. I had included only the ones that xbps-src requires at first, added these based on https://github.com/void-linux/void-packages/pull/40423#discussion_r1021732889.

Comment (paper42) [reply common/shlibs:3367]: ok, that's wrong, please revert it

Comment (ar-jan) [reply common/shlibs]: done

**Comment by paper42** [comment]
> The `-DVTK_MODULE_USE_EXTERNAL_vtknlohmannjson=OFF` configure arg did not help to avoid `CMake Warning at CMake/vtkModule.cmake:4572 (find_package): By not providing "Findnlohmann_json.cmake" in CMAKE_MODULE_PATH this project has asked CMake to find a package configuration file provided by "nlohmann_json", but CMake did not find one.`. Used `DVTK_USE_EXTERNAL=OFF` instead. I'm not sure if this is problematic.

We definitely want to use system dependencies, can we fix it by adding json-c++ to makedepends?

**Comment by kruceter** [comment]
> We definitely want to use system dependencies, can we fix it by adding json-c++ to makedepends?

Certainly.

Also fmt-devel and eigen are already present in the repository, so they are used in my case, too.

[vtk.txt](https://github.com/void-linux/void-packages/files/10048296/vtk.txt)
[vtk-query.txt](https://github.com/void-linux/void-packages/files/10048300/vtk-query.txt)

```diff
diff --git a/srcpkgs/vtk/template b/srcpkgs/vtk/template
index dfabbb649a..147718c690 100644
--- a/srcpkgs/vtk/template
+++ b/srcpkgs/vtk/template
@@ -6,21 +6,21 @@ build_style=cmake
 # vtk can be huge, especially with -DVTK_BUILD_ALL_MODULES=ON"
 # Build only the core modules plus python bindings for now
 configure_args="-DBUILD_SHARED_LIBS=ON -DVTK_FORBID_DOWNLOADS=ON
- -DVTK_USE_EXTERNAL=OFF
+ -DVTK_USE_EXTERNAL=ON
  -DVTK_MODULE_USE_EXTERNAL_VTK_utf8=OFF
  -DVTK_MODULE_USE_EXTERNAL_VTK_pegtl=OFF
  -DVTK_MODULE_USE_EXTERNAL_VTK_libharu=OFF
  -DVTK_MODULE_USE_EXTERNAL_VTK_exprtk=OFF
- -DVTK_MODULE_USE_EXTERNAL_VTK_fmt=OFF
- -DVTK_MODULE_USE_EXTERNAL_vtknlohmannjson=OFF
- -DVTK_MODULE_USE_EXTERNAL_vtkeigen=OFF
+ -DVTK_MODULE_USE_EXTERNAL_VTK_cgns=OFF
+ -DVTK_MODULE_USE_EXTERNAL_VTK_ioss=OFF
+ -DVTK_MODULE_USE_EXTERNAL_VTK_verdict=OFF
  -DVTK_WRAP_PYTHON=ON -DVTK_PYTHON_VERSION=3"
 # vtk forks libharu, bumps to v2.4.0, and requires libharu>=2.4.0
 makedepends="zlib-devel freetype-devel liblz4-devel expat-devel MesaLib-devel
  libXt-devel libjpeg-turbo-devel tiff-devel hdf5-devel netcdf-devel
  libxml2-devel jsoncpp-devel openmpi-devel libogg-devel libtheora-devel
  eigen double-conversion-devel glew-devel pugixml-devel sqlite-devel
- gl2ps-devel proj-devel python3-devel"
+ gl2ps-devel proj-devel python3-devel json-c++ fmt-devel"
 short_desc="System for 3D computer graphics, image processing, and visualization"
 maintainer="Piraty <piraty1@inbox.ru>"
 license="BSD-3-Clause"
```

**Review by kruceter — APPROVED** [review]

**Comment by ar-jan** [comment]
Bump. I just changed the hardcoded references to the vtk version, otherwise this is the same as before, and I think it's ready.

**Pull request merged**